openapi/components/schemas/Storefront/StorefrontTransaction.yaml

```
@@ -1,29 +1,280 @@
+          - invalid card/timeout
+          - unavailable
+      authenticated:
+        description: Authentication response status for 3D Secure.
+        type: string
+        enum:
+          - yes
+          - no
+          - not applicable
+          - attempted
+      liability:
+        type: string
+        enum:
+          - protected
+          - not protected
+          - protected (attempt)
+      flow:
+        description: Authentication flow for 3D Secure 2.
+        type: string
+        enum:
+          - frictionless
+          - challenge
+      isDowngraded:
+        description: Specifies if 3D Secure 2 is attempted and downgraded to 3D Secure 1.
+        type: boolean
+        default: false
+        deprecated: true
+  redirectUrl:
+    description: |-
+      URL where the end-user is redirected to when an offsite transaction is completed.
+      The default value is the website URL.
+    type: string
+    nullable: true
+    format: uri
+  retryNumber:
+    type: integer
+    readOnly: true
+    description: Position of the transaction in the sequence of retries.
+    x-sortable: true
+  isRetry:
+    type: boolean
+    readOnly: true
+    description: Specifies if a transaction is a retry.
+  billingDescriptor:
+    type: string
+    nullable: true
+    readOnly: true
+    description: |-
+      Billing descriptor that appears on the periodic billing statement.
+      For a credit card statement, this field commonly contains 12 or fewer characters.
+  description:
+    type: string
+    description: Description of the payment.
+    maxLength: 255
+  requestId:
+    description: |-
+      Request ID of the transaction. This ID must be unique within a 24 hour period.
+      Use this field to prevent duplicated transactions.
+    type: string
+    x-sortable: true
+  hasAmountAdjustment:
+    description: Specifies if the transaction has amount adjustment.
+    type: boolean
+    readOnly: true
+  gatewayName:
+    readOnly: true
+    description: |-
+      Name of the payment gateway that processed, or is selected to process, the transaction.
+      This value is only available after a gateway is selected for the transaction.
+    x-label: Gateway
+    x-basic: true
+    allOf:
+      - $ref: ../Gateways/GatewayName.yaml
+  customFields:
+    $ref: ../ResourceCustomFields.yaml
+  processedTime:
+    description: Date and time when the transaction is processed.
+    x-sortable: true
+    x-basic: true
+    allOf:
+      - $ref: ../ServerTimestamp.yaml
+  createdTime:
+    $ref: ../CreatedTime.yaml
+  updatedTime:
+    $ref: ../UpdatedTime.yaml
+  paymentInstrument:
+    $ref: ./PaymentInstruments/InstrumentReference.yaml
+  approvalUrl:
+    description: URL to redirect the end-customer when transaction `status` is `waiting-approval` or `offsite`.
+    type: string
+    format: uri
+  token:
+    description: Session's token used for authentication. It would allow to visit created order, invoice and transaction.
```

nitpick: [possessive apostrophes)](https://github.com/Rebilly/api-definitions/blob/main/WRITING-STYLE.md#possessive-apostrophes)
```suggestion
    description: Session token that is used for authentication. It would allow to visit created order, invoice and transaction.
```